aspnetcore/mvc/views/razor.md

```diff
@@ -512,92 +581,157 @@ If "rick@contoso.com" is passed in the model, the view generates the following H
 
-`@functions` methods serve as templating methods when they have markup:
+### \@namespace
 
-```cshtml
-@{
-    RenderName("Mahatma Gandhi");
-    RenderName("Martin Luther King, Jr.");
-}
+The `@namespace` directive:
 
-@functions {
-    private void RenderName(string name)
-    {
-        <p>Name: <strong>@name</strong></p>
-    }
-}
+* Sets the namespace of the class of the generated Razor page, MVC view, or Razor component.
+* Sets the root derived namespaces of a pages, views, or components classes from the closest imports file in the directory tree, *_ViewImports.cshtml* (views or pages) or *_Imports.razor* (Razor components).
+
+```cshtml
+@namespace Your.Namespace.Here
 ```
 
-The code renders the following HTML:
+For the Razor Pages example shown in the following table, the imports file at *Pages/_ViewImports.cshtml* contains `@namespace Hello.World`. The namespace of pages in the *Pages* folder that import the `Hello.World` namespace use `Hello.World` as the root of their derived namespaces. The same relationships apply to import files used with MVC views and Razor components.
 
-```html
-<p>Name: <strong>Mahatma Gandhi</strong></p>
-<p>Name: <strong>Martin Luther King, Jr.</strong></p>
-```
+| Page                                        | Namespace                             |
+| ------------------------------------------- | ------------------------------------- |
+| *Pages/Index.cshtml*                        | `Hello.World`                         |
+| *Pages/MorePages/Page.cshtml*               | `Hello.World.MorePages`               |
+| *Pages/MorePages/EvenMorePages/Page.cshtml* | `Hello.World.MorePages.EvenMorePages` |
+
+When multiple import files have a `@namespace` directive, the file closest to the page, view, or component in the directory tree is used to set the root namespace.
+
+If the *EvenMorePages* folder in the preceding example has an imports file with `@namespace Another.Planet` (or the *Pages/MorePages/EvenMorePages/Page.cshtml* file contains `@namespace Another.Planet`), the result is shown in the following table.
+
+| Page                                        | Namespace               |
+| ------------------------------------------- | ----------------------- |
+| *Pages/Index.cshtml*                        | `Hello.World`           |
+| *Pages/MorePages/Page.cshtml*               | `Hello.World.MorePages` |
+| *Pages/MorePages/EvenMorePages/Page.cshtml* | `Another.Planet`        |
+
+### \@page
+
+::: moniker range=">= aspnetcore-3.0"
+
+The `@page` directive has different effects depending on the type of the file where it appears. The directive:
+
+* Turns a file into an MVC action, which means that the file handles requests directly, without going through a controller. For more information, see <xref:razor-pages/index>.
```

Comment:
While this is technically true, but I think it would be simpler to say that the `@page` directive in a .cshtml file indicates that the file is a Razor Page. Is there a reason this wording was used?

Reply:
Taken from engineering language ... sounds kind'a *Ryan-y* to me. :smile: lol

I'll use your language.